### PR TITLE
add check for "rule->parname is not NULL" before using

### PR DESCRIPTION
**Pull request by red1452**
The PartitionRule structure can be allocated in the add_part_to_catalog
function, where the parname argument may be NULL. It means, that the parname
field at this structure will not be set. When this structure will be processed
at the get_partition_recursive function, this filed will be used without
checking. It can lead to segfault.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/63345

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035463

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035461

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035457

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035464

File: src/backend/utils/adt/ruleutils.c [hunk 1]
```diff
@@ -11117,7 +11117,7 @@ get_partition_recursive(PartitionNode *pn, deparse_context *head,
 					{
 						first_every_rule = NULL;
 
-						if (!strlen(rule->parname))
+						if (!rule->parname || !strlen(rule->parname))
```
Comment (RekGRpth) [inline src/backend/utils/adt/ruleutils.c]: How about to do same thing on similar [line](https://github.com/arenadata/gpdb/pull/698/files#diff-6c12827097e95912476267d73f3f56e8250b13aad37ab4286c3c7fdab7f30ec7R11013)?

**Comment by BenderArenadata** [comment]
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1035456

File: src/backend/utils/adt/ruleutils.c (hunk 1 above)
Comment (red1452) [reply src/backend/utils/adt/ruleutils.c]: done

**Review by RekGRpth — APPROVED** [review]

**Pull request merged**